### ISSUE TITLE
Referenced parameters not resolved

### ISSUE THREAD

**Issue by npdev453** [horiuchi/dtsgenerator#471]
There is an endpoint with two parameters: `id`, `name`. But `id` described as a $ref with path, that's described as supported here: [SupportedFeatures](https://github.com/horiuchi/dtsgenerator/blob/master/SupportedFeatures.md#openapi).

After running `dtsgen ./input.yaml` command return in `Parameters` namespace only one parameter — `name`.

> input.yaml
```
openapi: "3.0.0"
info:
  title: ParamRefExample
  version: 1.0.0
paths:
  "/endpoint/one/":
    get:
      operationId: operationOne
      parameters:
        - $ref: "#/components/parameters/id"
        - name: name
          in: query
          schema: 
            type: string
      responses:
        200:
          description: Success
components:
  parameters:
    id:
      required: true
      name: id
      in: query
      schema:
        type: string
```
> stdout
```
declare namespace Components {
    namespace Parameters {
        namespace Id {
            export type Id = string;
        }
    }
}
declare namespace Paths {
    namespace OperationOne {
        namespace Parameters {
            export type Name = string;
        }
        export interface QueryParameters {
            name?: Parameters.Name;
        }
    }
}
```

Based on: https://github.com/anttiviljami/openapi-client-axios/issues/47

**Comment by jschirrmacher** [comment]
I can confirm that parameter definitions in schema are not resolved correctly. I'm using v3.12.0 and get this error message:

`Error: The $ref targets root is not found: #/components/parameters/GeographyId`

My spec looks like that:
```
openapi: 3.0.3
info:
  title: Test
  version: "0.4.0"

paths:
  /geographies/{geographyId}:
    get:
      operationId: getGeography
      description: Get a single geography
      parameters:
        - $ref: "#/components/parameters/GeographyId"
      tags:
        - Geography
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Geography"

components:
  schemas:
    Geography:
      description: Information about a geography
      type: object
      properties:
        name:
          type: string
      example:
        name: Marketplace

  parameters:
    GeographyId:
      name: geographyId
      in: path
      description: unique id of geography
      required: true
      schema:
        type: string
        format: uuid
```

I use `npx dtsgen demo.yaml` and get

```
Error: The $ref targets root is not found: #/components/parameters/GeographyId
    at ReferenceResolver.resolve (/node_modules/dtsgenerator/dist/core/referenceResolver.js:77:19)
    at DtsGenerator.generate (/node_modules/dtsgenerator/dist/core/dtsGenerator.js:28:29)
    at async Object.dtsGenerator [as default] (/node_modules/dtsgenerator/dist/core/index.js:15:12)
    at async exec (/node_modules/dtsgenerator/dist/cli.js:111:20)
```

**Comment by Morl99** [comment]
As far as I can tell, this problem seems to be a regression introduced in 3.12.0, with 3.11.0 the problem is not reproducable for me, maybe this helps fixing it.

**Comment by jschirrmacher** [comment]
Yes, the root cause of this bug seems to be in #501

**Comment by horiuchi** [comment]
@jschirrmacher @Morl99 Thank you for your report of the bug.
I will investigate the reason.

**Comment by horiuchi** [comment]
Fixed and released by v3.12.1
